### ISSUE TITLE
deliver doesn't remove screenshots

### ISSUE THREAD

**Issue by lacostej** [fastlane/deliver#550]
1. run deliver to upload some screenshots (e.g. 5)
2. remove some from disk
3. re-run deliver

expected: removed screenshots on disk are removed from ITC

what happens: removed screenshots are not removed from ITC

**Comment by lacostej** [comment]
Current (spaceship based) work-around:
```
Spaceship.login(user, password)
Spaceship::Tunes.login(user, password)
du = Spaceship::Tunes.client.du_client
​
app = Spaceship::Tunes::Application.find bundle_id
​
appv = app.edit_version
​
appv.screenshots.each do |language, screenshots|
  screenshots.each do |sc|
    appv.upload_screenshot!(nil, sc.sort_order, sc.language, sc.device_type)
  end
end
​
appv.save!
```

**Comment by dermaaarkus** [comment]
i've got the same issue. we had 5 screenshots online and now i changed the project to use fastlane (thank your for the great work) and the last screenshot is not removed.
i do not use spaceship until now, is there any other solution?

thanks

**Comment by KrauseFx** [comment]
So `deliver` *should* delete the screenshots when there is at least one for this language/device. We don't want to remove existing screenshots if there are no new ones there.

**Comment by dermaaarkus** [comment]
this is the right way in my opinion :+1:

**Comment by ZackaryVink** [comment]
I am seeing this same issue. When I push 3 images (and there are already 4 on ITC), the three new images get put up but one of the old screenshots is still there.

**Comment by armadsen** [comment]
Seeing this too. My app was rejected for a problem with one screenshot. Being able to delete just that one by deleting it for all devices sizes and all localizations on disk then re-running deliver would be very useful, but it's not working.

**Comment by jeffreyjackson** [comment]
is it possible to delete all screenshots on iTC before upload?  deliver is only filling in the blanks, and i need to redo all screenshots completely.

**Comment by lacostej** [comment]
@jeffreyjackson right now it's not possible. I hope to be able to make deliver sync the local directory with the remote in the future. Right now, the simplest way I know to remove all screenshots is to use the spaceship script that I posted earlier in that issue.

**Comment by jeffreyjackson** [comment]
@lacostej do you know offhand if spaceship will need iTC credentials each time?  I'm looking to build this into a CI routine.

**Comment by jeffreyjackson** [comment]
@KrauseFx any suggestions for this?  maybe i can rally for a PR.

**Comment by jeffreyjackson** [comment]
I think this feature would also inherently fix #597
